helix-lock/src/main/java/org/apache/helix/lock/helix/ZKDistributedNonblockingLock.java

```diff
@@ -312,16 +312,20 @@ public ZNRecord update(ZNRecord current) {
         return _record;
       }
 
-      // higher priority lock request will try to  preempt current lock owner
-      if (!isCurrentOwner(curLockInfo) && _priority > curLockInfo.getPriority()) {
+      LockInfo unlockOrLockRequestLockInfo = new LockInfo(_record);
+      int unlockOrLockRequestPriority = unlockOrLockRequestLockInfo.getPriority();
+      // higher priority lock request will try to  preempt current lock owner. And any unlock request
+      // from non-lock owners will always be blocked because the default priority of unlock request
+      // is 0. And it would always be less than or equal to the priority value of the current lock.
+      if (!isCurrentOwner(curLockInfo) && unlockOrLockRequestPriority > curLockInfo.getPriority()) {
```

Comment:
This should work properly, but one thing is that unlock request will get the HelixException in the end of this method with text "failed to acquire lock", which can be confusing. I think it might be better to provide different exception message for each case or a general exception message that covers both cases.

Reply:
Make sense. Added a check that throws exception if the user is not the lock owner with descriptive messages.